From: Marcelo Soares Mota <motasmarcelo@gmail.com>
Date: Sat, 20 Feb 2021 21:14:12 -0300
Subject: fix failure in detect installed torbrowser

I believe i found an issue with the launcher (somehow related with #842021)

It happens that though the launcher works it doesn't detect torbrowser
as installed. It causes two issues:
  - The 'Settings' window points status as Not Installed;
  - The 'Browser launcher' keeps running the routines to download and
verify the 'first installation'.

I realized that reload the settings after rebuilding paths makes the
launcher works as expected.

Plus, load_settings() is not checking changes at 'installed' directive
to rewrite the settings file ('settings.json').

Closes: #983198
---
 torbrowser_launcher/common.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/torbrowser_launcher/common.py b/torbrowser_launcher/common.py
index e8e8d39..fb8ad50 100644
--- a/torbrowser_launcher/common.py
+++ b/torbrowser_launcher/common.py
@@ -64,6 +64,7 @@ class Common(object):
         self.load_mirrors()
         self.load_settings()
         self.build_paths()
+        self.load_settings()
         self.mkdir(self.paths["download_dir"])
         self.mkdir(self.paths["tbb"]["dir"])
         self.init_gnupg()
@@ -371,7 +372,10 @@ class Common(object):
             resave = False
 
             # detect installed
-            settings["installed"] = os.path.isfile(self.paths["tbb"]["start"])
+            installed = os.path.isfile(self.paths["tbb"]["start"])
+            if settings["installed"] != installed:
+                settings["installed"] = installed
+                resave = True
 
             # make sure settings file is up-to-date
             for setting in default_settings:
